### PR TITLE
Calculate the SemVer from Conventional Commits

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -28,14 +28,22 @@ import org.jetbrains.gradle.ext.settings
...
+
+    // Do not let untracked files bump the version or add a "-SNAPSHOT" suffix.
+    noDirtyCheck = true
+}
+
 // Only override a default version (which usually is "unspecified"), but not a custom version.
 if (version == Project.DEFAULT_VERSION) {
-    val gitVersionProvider = providers.of(GitVersionValueSource::class) { parameters { workingDir = rootDir } }
-    version = gitVersionProvider.get()
+    version = semver.version
```
Comment (mnonnenmacher) [inline build.gradle.kts:46]: Could you summarize the algorithm to calculate the version here (and maybe also in the commit message)?

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

File: build.gradle.kts (hunk 1 above)
Comment (sschuberth) [reply build.gradle.kts:46]: It's in the README, just [scroll down a bit](https://github.com/jmongard/Git.SemVersioning.Gradle?tab=readme-ov-file#versioning) from the link provided in the commit message.

Comment (sschuberth) [reply build.gradle.kts:46]: I've also added a brief description to the commit message.

**Review by mnonnenmacher — APPROVED** [review]
